src/pages/index.js

```diff
@@ -4,22 +4,103 @@ import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, validationConfig } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import { validationConfig } from "../utils/constants.js";
 import "../pages/index.css";
 
+// API Initialization
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "81e44564-32cf-4c2a-b8fb-18b8f71d8ac2",
+    "Content-Type": "application/json",
+  },
+});
+
 // Instantiate PopupWithImage
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
+// Delete Confirmation Popup
+const deleteConfirmPopup = new PopupWithConfirm(
+  "#delete-card-modal",
+  (cardId) => {
+    deleteConfirmPopup.renderLoading(true);
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        const cardElement = document.querySelector(
+          `[data-card-id="${cardId}"]`
+        );
+        if (cardElement) {
+          cardElement.remove();
+        }
+        deleteConfirmPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteConfirmPopup.renderLoading(false);
+      });
+  }
+);
+deleteConfirmPopup.setEventListeners();
+
+function handleDeleteButtonClick(cardId) {
+  deleteConfirmPopup.open(cardId);
+}
+
+function handleLikeButtonClick(card, cardId) {
+  const isLiked = card.isLiked();
+
+  const likeAction = isLiked ? api.dislikeCard(cardId) : api.likeCard(cardId);
+
+  likeAction
+    .then((updatedCard) => {
+      card.setLikes(updatedCard.isLiked);
+    })
+    .catch((err) => {
+      console.error("Like operation failed:", err);
+
+      if (card._likeButton) {
+        card._likeButton.disabled = false;
+      }
+    });
+}
+
+function handleImageClick(cardData) {
+  imagePopup.open({ name: cardData.name, link: cardData.link });
+}
+
+function createCard(cardData) {
+  const card = new Card(
+    {
+      ...cardData,
+      userId: userInfo.getUserInfo()._id,
+      owner: cardData.owner || { _id: cardData.ownerId },
+      isLiked: cardData.isLiked,
+    },
+    "#card-template",
+    handleImageClick,
+    handleDeleteButtonClick,
+    handleLikeButtonClick
+  );
+  return card.getView();
+}
+
 // DOM elements
+const profileImage = document.querySelector(".profile__image");
 const profileEditButton = document.querySelector("#profile-edit-button");
 const addNewCardButton = document.querySelector(".profile__add-button");
 const profileNameInput = document.querySelector("#profile-name-input");
 const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
+const profileImageContainer = document.querySelector(
+  ".profile__image-container"
+);
 
-// Instantiate form validators
+// Instantiate Form Validators
 const profileFormValidator = new FormValidator(
   validationConfig,
   document.querySelector("#profile-edit-form")
```

Comment:
**COULD BE IMPROVED**

We can get any form using their **unique** attribute `name` or `id` from `document.forms`

```js

const profileForm = document.forms["profile-form"];
const cardForm = document.forms["card-form"];

```

You need to add `name` or `id` first in `index.html` to achieve that.

More info here https://developer.mozilla.org/en-US/docs/Web/API/Document/forms